x-pack/plugins/observability_solution/entity_manager/server/lib/manage_index_templates.ts

```diff
@@ -23,6 +23,12 @@ interface ComponentManagementOptions {
   logger: Logger;
 }
 
+interface DeleteTemplateOptions {
+  esClient: ElasticsearchClient;
+  name: string;
+  logger: Logger;
+}
+
 export async function upsertTemplate({ esClient, template, logger }: TemplateManagementOptions) {
   try {
     await esClient.indices.putIndexTemplate(template);
```

Comment:
the templates/components methods were initially used in the plugin start that's probably why we swallowed errors. we're now using them at runtime so we should throw instead (does not harm to also throw on upsertComponent even if it's only used in plugin start)

Reply:
I update all methods on this file to throw the error.